website/docs/security.md

```diff
@@ -36,3 +36,181 @@ anonymous clients will be authenticated as user1,
+See how to [create user](https://www.postgresql.org/docs/current/sql-createuser.html).
+
+```sh
+mongosh `mongodb://user2:pass2@127.0.0.1/ferretdb?authMechanism=PLAIN`
+```
+
+#### Using docker
+
+For using docker, specify `FERRETDB_POSTGRESQL_URL` with default username and password.
+
+```yml
+services:
+  postgres:
+    image: postgres
+    environment:
+      - POSTGRES_USER=username
+      - POSTGRES_PASSWORD=password
+      - POSTGRES_DB=ferretdb
+    volumes:
+      - ./data:/var/lib/postgresql/data
+
+  ferretdb:
+    image: ghcr.io/ferretdb/ferretdb
+    restart: on-failure
+    ports:
+      - 27017:27017
+    environment:
+      - FERRETDB_POSTGRESQL_URL=postgres://user1:pass1@postgres:5432/ferretdb
+
+networks:
+  default:
+    name: ferretdb
+```
+
+To start `ferretdb`, use docker compose.
+
+```sh
+docker compose up -d
+```
+
+An anonymous client is authenticated with `user1` from `FERRETDB_POSTGRESQL_URL`.
+Use following command to run `mongosh` inside the temporary MongoDB container,
+attached to the same Docker network.
+
+```sh
+docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo `mongodb://ferretdb/ferretdb`
+```
+
+Clients that specify username and password in MongoDB URI such as below is authenticated as `user2`.
+
+```sh
+docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo \
+`mongodb://user2:pass2@ferretdb/ferretdb?authMechanism=PLAIN`
+```
+
+### PostgreSQL backend with TLS
+
+Using TLS is recommended if username and password are transferred in plain text.
+
+In following examples, FerretDB uses TLS certificates to secure the connection.
+Example certificates are found in [build/certs](https://github.com/FerretDB/FerretDB/tree/main/build/certs).
+The server uses TLS server certificate file, TLS private key file and root CA certificate file.
+
+```console
+server-certs/
+├── rootCA-cert.pem
+├── server-cert.pem
+└── server-key.pem
+```
+
+The client uses TLS client certificate file and root CA certificate file.
+
+```console
+client-certs/
+├── client.pem
+└── rootCA-cert.pem
+```
+
+#### Using TLS with `ferretdb` package
+
+The example below connects to localhost PostgreSQL instance using TLS with certificates in `server-certs` directory.
+Be sure to check that `server-certs` directory and files are present.
+
+```sh
+ferretdb \
+--postgresql-url=postgres://user1:pass1@localhost:5432/ferretdb \
+--listen-tls=:27018 \
+--listen-tls-cert-file=./server-certs/server-cert.pem \
+--listen-tls-key-file=./server-certs/server-key.pem \
+--listen-tls-ca-file=./server-certs/rootCA-cert.pem
+```
+
+Using `mongosh`, the client connects to ferretdb as `user2` using TLS certificates in `client-certs` directory.
+Be sure to check that `client-certs` directory and files are present.
+
+```sh
+mongosh 'mongodb://user2:pass2@127.0.0.1:27018/ferretdb?authMechanism=PLAIN&tls=true&tlsCertificateKeyFile=./client-certs/client.pem&tlsCaFile=./client-certs/rootCA-cert.pem'
+```
+
+#### Using TLS with docker
```

Comment:
I think it is strange that all that is in the top-level section "Authentication" and in not in "Securing connections with TLS". I think some re-arrangement is needed. @Fashander PTAL

Reply:
Thanks for this, having it under `Securing connections with TLS` is more appropriate. Also @Fashander if you have other rearranging suggestions, please let me know!

Reply:
We should separate that into their own sections with security as the main header

``` 
- Security
-----Authentication
-----Securing with TLS

...
```

Reply:
https://docs.ferretdb.io/security/ – we already have those two sections. Please suggest more specific re-org

Reply:
Okay. 

I'm suggesting that rather than have them all in a single `.md` page now, we should have them all in a folder `Security`. This will probably be better for navigation, plus the section will probably still expand in the future.

Instead, we should have the `Securing connections with TLS` in single `.md doc.`, with `Authentication` in another. A more high-level view of this would be:

```
# Security (Folder/File name)

## Authentication (single doc)
### PostgreSQL backend with default username and password
#### Using ferretdb package
#### Using docker


## Securing connections with TLS (single doc)
### PostgreSQL backend with TLS
#### Using TLS with ferretdb package
#### Using TLS with docker
```

If we are still using the current single-doc, then it can be restructured like the one above.

Reply:
Please see the new format as suggested.